chainerx/_cuda.py

```diff
@@ -0,0 +1,38 @@
+import chainerx
+
+try:
+    # _pybind_cuda is unavailable if ChainerX is built without CUDA.
+    from chainerx import _pybind_cuda
+    _available = True
+except Exception:
+    _available = False
+
+try:
+    import cupy
+    _cupy_available = True
+except Exception:
+    _cupy_available = False
+
+_chainerx_allocator = None
```

Comment:
I think this variable is not needed.

Reply:
Must not the instance of `CFunctionAllocator` be kept alive? Only its `malloc` is passed to `set_allocator`.

Reply:
I think it's kept alive via the bound method `malloc`.

Reply:
Ah it seems to work. Fixed.